utils/events.go

```diff
@@ -66,3 +67,42 @@ func GetEventsValue(events types.StringEvents, t string, key string) string {
 	}
 	return ""
 }
+
+func ParseCoinFromEventString(coinStr string) (uint64, error) {
+	coin, err := types.ParseCoinNormalized(coinStr)
+	if err != nil {
+		return 0, err
+	}
+	return coin.Amount.Uint64(), nil
+}
+
+func GetRoyaltyMap(events types.StringEvents) map[string]uint64 {
+	royaltyMap := make(map[string]uint64)
+	for _, event := range events {
+		if event.Type == "coin_received" {
+			stakeholder := ""
+			amount := uint64(0)
+			for _, attr := range event.Attributes {
+				if attr.Key == "receiver" {
+					stakeholder = attr.Value
+				}
+				if attr.Key == "amount" {
+					amountStr := attr.Value
+					coin, err := types.ParseCoinNormalized(amountStr)
+					if err != nil {
+						logger.L.Warnw("Failed to parse royalty from event", "royalty_str", amountStr, "error", err)
+						stakeholder = ""
+						continue
+					}
+					amount = coin.Amount.Uint64()
+				}
+				if stakeholder != "" && amount != 0 {
+					royaltyMap[stakeholder] = amount
```

Comment:
what if the receiver is both the owner of the nft and a stakeholder? Or if the stakeholder is defined multiple times in royalty config?
Maybe we should use sth like `royaltyMap[stakeholder] += amount`
Also can we distinguish if the address is receiving LIKE due to it is stakeholder or owner?